projects/observable-webworker/src/lib/from-worker-pool.ts

```diff
@@ -0,0 +1,70 @@
+import { EMPTY, Observable, of, Subject } from 'rxjs';
+import { finalize, mergeMap } from 'rxjs/operators';
+import { fromWorker } from './from-worker';
+
+export function fromWorkerPool<I, O>(
+  workerConstructor: (index: number) => Worker,
+  workUnitIterator: IterableIterator<I> | Array<I>,
```

Comment:
For symmetry with fromWorker, allow this to be Observable too, and just buffer internally into an iterator

Reply:
I think this is not safe.

What happens when we call `.next()` on the iterator and there hasn't been an emission on the observable?

Reply:
Async iterators to the rescue. We can await iterator.next()

Will need to manage the  unsubscribe such that unsubscribing from the fromWorkerPool unsubs the input too

Reply:
Maybe just zipping the input stream with the idle stream will make life way easier?

Reply:
Sure did! https://github.com/cloudnc/observable-webworker/pull/17